### PR TITLE
Adding Cycle Sort in JavaScript

### PR DESCRIPTION
**Pull request by ireneelizabethsabu**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description
Adding cycleSort.js file
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
This PR is related to #2505
(Write your answer here.)

### CODE REVIEW

**Comment by ireneelizabethsabu** [comment]
@HarshCasper please review this PR.

File: JavaScript/sort/cycleSort.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+let input = [1, 8, 7, 61, 5, 4, 11];
+cycleSort(input);
+console.log(input);
+//EXPECTED: [1,4,5,7,8,11,61]
+
+//CASE 2:
+input = [1,8,3,9,10,10,2,4];
+cycleSort(input)
+console.log(input);
+//EXPECTED: [1,2,3,4,8,9,10,10]
```
Comment (HarshCasper) [inline JavaScript/sort/cycleSort.js:63]: Use Dynamic Input

File: JavaScript/sort/cycleSort.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+/*
+CYCLE SORT:
+    Selection sort is a inplace sorting algorithm which optimises the number of memory writes.
```
Comment (HarshCasper) [inline JavaScript/sort/cycleSort.js:3]: Instead focus this part on explaining the algorithm

File: JavaScript/sort/cycleSort.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    Time Complexity - O(n^2)
+    Space Complexity - O(1)
```
Comment (HarshCasper) [inline JavaScript/sort/cycleSort.js:5]: Shift Time and Space Complexity at the end of the Code

File: JavaScript/sort/cycleSort.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+*/
+
+function cycleSort(arr) {
```
Comment (HarshCasper) [inline JavaScript/sort/cycleSort.js:8]: Prefer using ES6 conventions

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Pull request closed without merging**